#!/bin/bash

# this script will run deepbinner with basecalled reads
# for more information read https://github.com/rrwick/Deepbinner#using-deepbinner-after-basecalling

threads=20
fast5dir="20180209_2242_20180209_hsalinarum_genomes_run/fast5"
basecalleddir="20180816_basecalled/workspace"
deepbinnerdir="20180817_deepbinner"

if [[ ! -d $deepbinnerdir ]] ; then mkdir $deepbinnerdir ; fi

# first we need to merge the fastq files generated by albacore
find $basecalleddir -name "*.fastq" -exec cat {} + > ${deepbinnerdir}/all_basecalled_reads.fastq

# classifying reads
deepbinner classify -s /opt/Deepbinner/models/EXP-NBD103_read_starts \
                    -e /opt/Deepbinner/models/EXP-NBD103_read_ends \
		    --intra_op_parallelism_threads $threads \
		    --omp_num_threads $threads \
		    --verbose \
		    $fast5dir > ${deepbinnerdir}/classifications 2> ${deepbinnerdir}/deepbinner_classify.err

# binning fastqreads
deepbinner bin --classes ${deepbinnerdir}/classifications \
	       --reads ${deepbinnerdir}/all_basecalled_reads.fastq \
	       --out_dir ${deepbinnerdir} > ${deepbinnerdir}/deepbinner_bin.log 2>&1 


